src/thingset_txt.c

```diff
@@ -550,9 +550,17 @@ int ts_txt_patch(struct ts_context *ts, const struct ts_data_object *parent)
 
         // extract the value and check buffer lengths
         value_len = ts->tokens[tok].end - ts->tokens[tok].start;
-        if ((object->type != TS_T_STRING && value_len >= sizeof(value_buf)) ||
-            (object->type == TS_T_STRING && value_len >= (size_t)object->detail))
-        {
+        if (object->type == TS_T_STRING) {
+            if (value_len < (size_t)object->detail) {
```

Comment:
Does this account for the trailing '\0' that will be appended by strncpy?

Reply:
I think so. The detail contains the `sizeof(str_buffer)`. As it checks for `<` and not `<=` this should work. Let's take an example:

```C
char buf[4];
size_t detail = sizeof(buf); // gives 4
char json_str[] = "\"test\":\"str\"";
size_t value_len = 3; // str
assert(value_len < detail);
// this would be fine, but it would fail for a string token length of 4, 
// which does not fit into the buffer incl. the trailing zero
```

Reply:
You are right.